Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -136,22 +143,52 @@ Vector<StyledMarkedText> StyledMarkedText::subdivideAndResolve(const Vector<Mark
         return { styledMarkedText };
     }
 
-    auto markedTexts = MarkedText::subdivide(textsToSubdivide);
+    auto markedTexts = MarkedText::subdivide(textsToSubdivide, OverlapStrategy::None);
     ASSERT(!markedTexts.isEmpty());
     if (UNLIKELY(markedTexts.isEmpty()))
         return { };
 
+    if (!markedTexts.isEmpty()) {
+        // Get rid of duplicates in vector
+        ListHashSet<MarkedText> noDuplicates;
+        for (auto& elem : markedTexts)
+            noDuplicates.add(elem);
+        // Add back to markedTexts
+        if (!noDuplicates.isEmpty()) {
+            markedTexts.clear();
+            for (auto& elem : noDuplicates)
+                markedTexts.append(elem);
+        }
+        // Sort so highlights follow correct priority/insertion order
+        if (!markedTexts[0].highlightName.isNull()) {
+            // Maintain copy of old unordered markedTexts
+            Vector<MarkedText> dummy;
+            dummy = markedTexts;
+            markedTexts.clear();
+            for (auto& hiliteName : markedTextsNames) {
+                for (auto& elem : dummy) {
+                    if (elem.highlightName == hiliteName)
+                        markedTexts.append(elem);
+                }
+            }
+            // Make sure selection shows on top
+            for (auto& elem : dummy) {
+                if (elem.highlightName.isNull())
+                    markedTexts.append(elem);
+            }
+        }
+    }
+
     // Compute frontmost overlapping styled marked texts.
-    frontmostMarkedTexts.reserveInitialCapacity(markedTexts.size());
-    frontmostMarkedTexts.uncheckedAppend(resolveStyleForMarkedText(markedTexts[0], baseStyle, renderer, lineStyle, paintInfo));
+    frontmostMarkedTexts.append(resolveStyleForMarkedText(markedTexts[0], baseStyle, renderer, lineStyle, paintInfo));
```

Comment:
is this change needed?

Reply:
I changed it bc when testing out diff implementations, at one point the markedTexts could've been empty which caused the browser to crash. Testing now with the original uncheckedAppend and it works so the change isn't needed anymore.